src/js/chat.js

```diff
@@ -418,143 +437,219 @@ $( document ).ready(function() {
+                    console.log("Authenticated");
+                    sock.send("JOIN #" + CHANNEL + "\r\n");
+                    break;
+                case "JOIN":
+                    failedAttempts = 0;
+
+                    //$("#chat-content").css("background-color", "rgba(0,0,0,0)");
+                    var nick = cmd.origin.split("!")[0];
+                    if (nick == NICK) {
+                        console.log("Joined " + cmd.params[0]);
+                        console.log("Loading history...");
+                        $.get("http://irc.eternagame.org:8082/history.html", function (data) {
+                            console.log("History recieved");
+                            var messages = data.trim().split("\n");
+                            var firstNewMessage = 0;
+                            var j;
+                            for (j = 0; j < messages.length; j++)
+                                if (postedMessages.indexOf(messages[j].trim()) == -1) {
+                                    break;
+                                }
 
-sock.onmessage = function (e) {
-    var commands = parseCommands(e.data);
-    for (var i=0; i<commands.length; i++) {
-        var cmd = commands[i];
-        switch(cmd.command) {
-            case "PING":
-                sock.send("PONG :0.0.0.0\r\n");
-                break;
-            case "433":
-            // Nick already used, try with fallback
-                var nickNum = parseInt(NICK.match(/\^(\d+)/)[1]) + 1;
-                NICK = NICK.replace(/\^(\d+)/, "^" + nickNum);
-                sock.send("NICK " + NICK + "\r\n");
-                sock.send("USER " + "anon" + " 0 * :" + USERNAME + "\r\n");
-                break;
-            case "001":
-            // Initial info
-                console.log("Authenticated");
-                sock.send("JOIN #" + CHANNEL + "\r\n");
-                break;
-            case "JOIN":
-                var nick = cmd.origin.split("!")[0];
-                if (nick == NICK) {
-                    console.log("Joined " + cmd.params[0]);
-                    console.log("Loading history...");
-                    $.get( "http://irc.eternagame.org:8082/history.html", function( data ) {
-                        console.log("History recieved");
-                        var messages = data.trim().split("\n");
-                        for (var j=0; j<messages.length; j++) {
-                            postMessage(messages[j], true);
+                            if (!firstConnection)
+                                postMessage("Reconnected to chat - Some messages might be missing if you were away for a long time", true);
+                            for (; j < messages.length; j++) {
+                                postMessage(messages[j], true);
+                            }
+                            while (toBePosted.length)
+                                postedMessages(toBePosted.shift(), true);
```

Comment:
You mean `postMessage()`

Reply:
I thought I tested it... maybe without someone posting while it is reconnecting? Will test it fully before repushing

Reply:
That would be good, thanks